addon/mixins/dom.js

```diff
@@ -141,7 +141,22 @@ export default Mixin.create({
 
   willDestroyElement() {
     this._super(...arguments);
+    this._removeEventListeners();
+    // Determines whether or not we need to perform cleanup in the willDestroy hook
+    this._hasCalledWillDestroyElement = true;
+  },
 
+  willDestroy() {
+    this._super(...arguments);
+    if (!this._hasCalledWillDestroyElement) {
```

Comment:
I think we can just call _removeEventListeners again, and it will essentially do nothing...